.github/workflows/test-full-sync.yml

```diff
@@ -5,7 +5,9 @@ on:
     inputs:
       network:
         default: 'Mainnet'
-  pull_request_review:
+  # only run on Mergify head branches:
+  # https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#running-your-workflow-based-on-the-head-or-base-branch-of-a-pull-request-1
```

Comment:
```suggestion
```

Having the comment here might be confusing. Just inside the job should be enough to understand the `if` statement

Reply:
Our priority is getting CI stable and working, so I'd like to focus on this specific bug fix in this PR.

These questions are about functionality changes. Can you open discussions or tickets for these questions?

Once we have fixed this bug, and the test is stable, we'll have better information to make these decisions. Then we can schedule those changes in a sprint. (I'd like to check with the team before we make changes with a big impact on CI performance or failure rates.)

> > Shouldn't we consider adding `Test full Mainnet sync` job to Mergify's configuration file as a required check?

We can't make this change right now, because the job is currently failing. It took more than 6 hours on the most recent PR.

> Once upon a time we said we wanted the full sync test to always run, but not block merge, because it takes a long time. Is that still what we want?

That is the current configuration, and it will continue to be the configuration after this PR merges.

I think our decision will depend on how fast and reliable the full sync test is.

Reply:
I was confused about how often this workflow run, because the full filters aren't documented at the top of the file.

But this fix is urgent, so I'm happy to remove the extra docs if you want.